task/07-yield-tasks.js

```diff
@@ -145,7 +210,17 @@ function* mergeSortedSequences(source1, source2) {
  *   Most popular implementation of the logic in npm https://www.npmjs.com/package/co
  */
 function async(generator) {
-    throw new Error('Not implemented');
+    let newGenerator = generator();
+
+    return Promise.resolve((function func(val){
```

Comment:
split the definition of the func function and the usage of this function. Use better name for "func"